### PR TITLE
Makes the role stateless.

### LINKED ISSUE #18: Need for storing state locally?
Hi,

thanks for writing and sharing this role, this is really nice!

I have a question, is it necessary to store the state locally?
It would be easy to check if the server is already configured, find the private key and derive the public key, and if not, create the private key?

Am I missing something?

If you agree that storing the state is not so necessary, I can PR, to remove this need.

Cheers!

### PR DESCRIPTION
**Pull request by pierreozoux**
A lot of code removed ;)

closes #18

### CODE REVIEW

**Comment by pierreozoux** [comment]
Hello!

What do you prefer?

Do you prefer to merge PR one by one? Or do you prefer I create a big PR with all changes in different commits?

I'd like to go further on this topic, but if you can't merge right now, I'll adopt a different strategy.
No stress :) I'd like to avoid the free software burn out ;)

Cheers!

Pierre

**Comment by githubixx** [comment]
Hi! TBH, I have no idea :wink: Normally Smaller changes are way easier to review. But of course it always depends. So I would say make it as small as possible and as big as necessary :smile: I'll try to catch up with the PRs today or tomorrow. But I would also like to develop kind of a upgrade/migration guide for current users if possible. And since I'm a current user I hopefully be able to create one. I guess it will hurt anyways but maybe it's possible to reduce the pain a little bit...

**Comment by githubixx** [comment]
This PR currently has merge conflicts because of https://github.com/githubixx/ansible-role-wireguard/pull/27. But should be easy to adjust.

**Comment by pierreozoux** [comment]
I'll adjust today, thanks for your time :)

**Comment by pierreozoux** [comment]
Ok, adujsted.

**Comment by githubixx** [comment]
I was not sure at the beginning but since this is already a breaking change we should also include the changes to support multiple Wireguard interfaces as discussed in https://github.com/githubixx/ansible-role-wireguard/issues/25 + the Molecule tests for which you already created the PR  https://github.com/githubixx/ansible-role-wireguard/pull/22. Then we've all in one which makes it easier to test. I hope this works for you.

File: defaults/main.yml [hunk 1]
```diff
@@ -1,16 +1,4 @@
...
-# (the user that run's "ansible-playbook" command) plus
-# "/wireguard/certs". That means if the user's ${HOME} directory is e.g.
-# "/home/da_user" then "wireguard_cert_directory" will have a value of
-# "/home/da_user/wireguard/certs". If you change this make sure that
-# the parent directory is writable by the user that runs "ansible-playbook"
-# command.
-wireguard_cert_directory: "{{ '~/wireguard/certs' | expanduser }}"
-wireguard_cert_owner: "root"
-wireguard_cert_group: "root"
-
```
Comment (githubixx) [inline defaults/main.yml]: This part also needs to be removed from README.md in the `Role Variables` section.

Comment (githubixx) [reply defaults/main.yml]: Also the CHANGELOG.md needs to be updates. I guess when we ready with this PR we need to adjust it a little bit again but for now I guess we can already add this text:

```
**4.0.0**

- IMPORTANT: Version 4.0.0 has some breaking changes and is no longer compatible with versions < 4.0.0!
- Role is now stateless. It doesn't store private and public keys anymore on your local machine. Therefore the variables `wireguard_cert_directory`, `wireguard_cert_owner` and `wireguard_cert_group` are now obsolete and no longer needed (contribution by pierreozoux)
```
If you merge the code for supporting multiple Wireguard interfaces into this PR then the CHANGELOG text above needs to be extended e.g.:
```
- Role now supports multiple Wireguard interfaces. README was updated accordingly (contribution by pierreozoux)
```
So please also update README accordingly when you implement https://github.com/githubixx/ansible-role-wireguard/issues/25.

File: tasks/main.yml [hunk 2]
```diff
@@ -21,114 +21,68 @@
...
+- name: Check if WireGuard file exists
+  stat:
+    path: "{{ wg_conf_path }}"
+  register: wg_conf
 
-- name: Set path to public key file
-  set_fact:
-    public_key_file_path: "{{ wireguard_cert_directory }}/{{ inventory_hostname }}.public.key"
+- name: Read WireGuard private key
+  shell: "cat {{ wg_conf_path }} | grep PrivateKey | cut -d= -f2- | tr -d [:space:]"
```
Comment (githubixx) [inline tasks/main.yml]: `sed` should be good enough:
```
shell: "sed -n 's/PrivateKey = \(.*\)/\1/p' {{ wg_conf_path }}"
```

Comment (githubixx) [reply tasks/main.yml]: Maybe even `command` module may work:
```
- name: Read WireGuard private key
  command:
    argv:
      - sed
      - -n
      - 's/PrivateKey = \(.*\)/\1/p'
      - {{ wg_conf_path }}
```

**Comment by githubixx** [comment]
Functionality implemented with https://github.com/githubixx/ansible-role-wireguard/pull/29 .

**Pull request closed without merging**